recipes/MEDIA/media_prepare.py

```diff
@@ -0,0 +1,910 @@
+                sync_waiting,
+                has_speech,
+                concept,
+                concept_open,
+                n,
+                time,
+                time_end,
+            )
+
+        if node.nodeName == "Sync":
+            sentences, has_speech, sync_waiting, time, n = process_sync_node(
+                node,
+                sentences,
+                sync_waiting,
+                has_speech,
+                concept_open,
+                task,
+                n,
+                time,
+                time_end,
+            )
+
+    sentences = clean_last_sentence(sentences, n)
+
+    return sentences
+
+
+def process_text_node(
+    node,
+    sentences,
+    sync_waiting,
+    has_speech,
+    concept,
+    concept_open,
+    task,
+    n,
+    time_end,
+):
+    # Add a new concept, when speech following
+    if task == "slu" and concept != "null" and not concept_open:
+        sentences[n][0] += "<" + concept + "> "
+        sentences[n][1] += "<" + concept + "> _ "
+        concept_open = True
+    sentence = normalize_sentence(node.data)
+    sentences[n][0] += sentence + " "
+    sentences[n][1] += (
+        " ".join(list(sentence.replace(" ", "_"))) + " _ "
+    )
+    sentences[n][3] = time_end
+    has_speech = True
+    sync_waiting = False
+    return sentences, has_speech, sync_waiting, concept_open
+
+
+def process_sync_node(
+    node,
+    sentences,
+    sync_waiting,
+    has_speech,
+    concept_open,
+    task,
+    n,
+    time,
+    time_end,
+):
+    # If the segment has no speech yet
+    if not (has_speech):
+        # Change time_beg for the last segment
+        sentences[n][2] = node.getAttribute("time")
+    # If the segment has speech and sync doesn't cut a concept
+    elif task == "asr" or (task == "slu" and not concept_open):
+        # Change time_end for the last segment
+        sentences[n][3] = node.getAttribute("time")
+        sentences.append(["", "", sentences[n][3], time_end])
+        has_speech = False
+        n += 1
+    else:
+        sync_waiting = True
+        time = node.getAttribute("time")
+    return sentences, has_speech, sync_waiting, time, n
+
+
+def process_semfin_node(
+    sentences,
+    sync_waiting,
+    has_speech,
+    concept,
+    concept_open,
+    n,
+    time,
+    time_end,
+):
+    # Prevent adding a closing concept
+    # If Sync followed by SemFin generate a new segment without speech yet
+    if concept_open:
+        sentences[n][0] += "> "
+        sentences[n][1] += "> _ "
+    concept = "null"  # Indicate there is no currently open concept
+    concept_open = False
+    if sync_waiting:
+        sentences[n][3] = time
+        sentences.append(["", "", time, time_end])
+        has_speech = False
+        sync_waiting = False
+        n += 1
+    return sentences, concept, concept_open, has_speech, sync_waiting, n
+
+
+def clean_last_sentence(
+    sentences,
+    n,
+):
+    for n in range(len(sentences)):
```

Comment:
you are overriding the variable `n`, could you please explain what is the purpose of the `n` in the function parameters?